wrappers/java/src/main/java/org/hyperledger/indy/sdk/agent/Agent.java

```diff
@@ -0,0 +1,329 @@
+package org.hyperledger.indy.sdk.agent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+import org.hyperledger.indy.sdk.LibSovrin;
+import org.hyperledger.indy.sdk.SovrinException;
+import org.hyperledger.indy.sdk.SovrinJava;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentAddIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseConnectionResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseListenerResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentConnectResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentListenResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentRemoveIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentSendResult;
+import org.hyperledger.indy.sdk.pool.Pool;
+import org.hyperledger.indy.sdk.wallet.Wallet;
+
+import com.sun.jna.Callback;
+
+/**
+ * agent.rs API
+ */
+public class Agent extends SovrinJava.API {
+
+	private Agent() {
+
+	}
+
+	/*
+	 * STATIC METHODS
+	 */
+
+	public static Future<AgentConnectResult> agentConnect(
```

Comment:
I see that all return types have Result wrapper. For me, it looks like just additional boilerplate. I suggest:

1. if there are no fields in Result then just return Future<Void>
2. if we return an intermediate object like AgentConnection then return it directly without Result wrapper. For example,

`public static Future<AgentConnection> connect`

3. If the result is a simple type like String then use this type directly. For example, `Result<String>`
4. And only If the result is a tuple then define a custom Result type.